File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    const temp = sizes.map(([w, h]) => w < h ? [h, w] : [w, h]);
    let maxW = 0;
    let maxH = 0;
    for (const [w, h] of temp) {
        if (maxW < w) maxW = w;
        if (maxH < h) maxH = h;
    }
    return maxW * maxH;
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;
class Solution {
    public int solution(int[][] sizes) {
        int maxWidth = 0;
        int maxHeight = 0;
        for (int[] size : sizes) {
            int max = Math.max(size[0], size[1]);
            int min = Math.min(size[0], size[1]);
            maxWidth = Math.max(maxWidth, max);
            maxHeight = Math.max(maxHeight, min);
        }
        return maxWidth * maxHeight;
    }
}